subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsBuildOperationResult.java

```diff
@@ -149,6 +192,9 @@ public void enterDirectory(DirectorySnapshot physicalSnapshot) {
         @Override
         public SnapshotVisitResult visitEntry(FileSystemLocationSnapshot snapshot) {
             if (snapshot.getType() == FileType.Directory) {
+                if (propertyAttributes.contains(PropertyAttribute.DIRECTORY_SENSITIVITY_IGNORE_DIRECTORIES) && ((DirectorySnapshot) snapshot).getChildren().isEmpty()) {
+                    return SnapshotVisitResult.SKIP_SUBTREE;
```

Comment:
didn't find a cleaner way than casting the snapshot right here :/

Reply:
I'd actually pull this condition up:
```
            FileSystemLocationFingerprint fingerprint = fingerprints.get(path);
            if (fingerprint == null) {
                return SnapshotVisitResult.CONTINUE;
            }
```

If we ignore empty directories, we ignore all directories. So they won't be part of the fingerprint. Your solution for example would still take a directory which contains an empty directory into account. We don't need to skip the subtree.